### DISCUSSION TITLE
Eframe template only renders once on windows

### DISCUSSION THREAD

**Discussion by Lord-Hellgrim** [emilk/egui#3764] [Q&A]
Tl;dr: Eframe template does not display any window but runs logic and captures events.

The first time I compile and run a raw eframe template (meaning I have not changed the code at all) it runs fine. The second time I run it, it doesn't generate any window but does seem to be running the framework since print statements run just fine, even accurately capturing keypresses and so on. I am running windows 11. This is super weird and is probably just me being stupid, but I can't figure out what's wrong. I've checked if the window is rendering offscreen or something and that does not seem to be the case. There are no error messages to report either, it just silently doesn't render anything. Does anyone have any idea what's wrong?

**Comment by emilk** [comment]
Curious!

It is probably because of persistence. If you run with `RUST_LOG=debug` you should see a line `Loading app state from …" pointing to where eframe persist window position, size, and so on. Maybe something is going wrong there. If you open up the `app.ron` file it should give you some clues.

What is your monitor setup? Multiple monitors? Are they dpi scaling?

**Comment by Lord-Hellgrim** [comment]
I am just running on a laptop. No fancy monitor setup. I'll look into
persistence and get back to you. Thanks!

**Reply by Lord-Hellgrim** [reply]
Your diagnosis was correct. Here is the relevant text of the app.ron file:

"window": "(inner_position_pixels:Some((x:4713.0,y:2032.0)),outer_position_pixels:Some((x:4695.0,y:1947.0)),fullscreen:false...

If I manually set the inner and outer positions, the window renders correctly, but if I close and reopen the app, it resets the position to this out of bounds location. What could be causing this? I am willing to dig into the codebase and fix the underlying issue if you can point me in the right direction.

**Reply by Lord-Hellgrim** [reply]
Solved by disabling persistence. I will implement my own persistence features if needed. Thank you for your assistance and let me know if you can point me in the right direction to fix the underlying issue.

**Reply by Lord-Hellgrim** [reply]
I've been digging a bit and it seems this is a windows syscall giving the wrong window position to winit. Will investigate further.

**Reply by Lord-Hellgrim** [reply]
Ok, the windows syscall is returning correctly but something is moving the start position of the window by a strange amount every time. Here is the progression of screen co-ordinates (x, y):
on the first run I had deleted the app.ron file. Every subsequent run has an app.ron file.
90, 157
234, 301
666, 733
1962, 2029
5850, 3577
7938, 3577

Very strange. Will investigate further.

**Reply by Lord-Hellgrim** [reply]
Have been struggling for a while. Can't figure it out. Won't investigate further. Will just turn off persistence and implement my own if needed. I am unable to fix the bug at the library level. I am sorry.
